.factory/tests/test_upstream_sync_check.py

```
@@ -0,0 +1,127 @@
+            ["bash", str(dn / ".factory/upstream-sync-check.sh"), *args],
+            cwd=dn, env=git_env(), capture_output=True, text=True,
+        )
+
+    def test_clean_reports_no_action(self, synced):
+        _, dn, calls = synced
+        proc = self._run(dn)
+        assert proc.returncode == 0, proc.stdout + proc.stderr
+        assert "full 面干净，无动作" in proc.stdout
+        lines = calls.read_text().splitlines()
+        assert lines, "凭据探测须真实发生"
+        assert all("auth" in l for l in lines), f"干净基态不得触发远端命令: {lines}"
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)